chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationResultDetailsDAO.java

```diff
@@ -21,6 +22,28 @@
 public class CertificationResultDetailsDAO extends BaseDAOImpl {
     private static final Logger LOGGER = LogManager.getLogger(CertificationResultDetailsDAO.class);
 
+    private static final String BASE_SQL = "SELECT DISTINCT crd "
+            + "FROM CertificationResultDetailsEntity crd "
+            + "LEFT OUTER JOIN FETCH crd.certificationCriterion cc "
+            + "LEFT OUTER JOIN FETCH cc.certificationEdition ed "
+            + "LEFT OUTER JOIN FETCH crd.certificationResultTestData crtd "
+            + "LEFT OUTER JOIN FETCH crtd.testData td "
+            + "LEFT OUTER JOIN FETCH crd.certificationResultTestFunctionalities crtf "
+            + "LEFT OUTER JOIN FETCH crtf.testFunctionality tf "
+            + "LEFT OUTER JOIN FETCH crd.certificationResultTestProcedures crtp "
+            + "LEFT OUTER JOIN FETCH crtp.testProcedure tp "
+            + "LEFT OUTER JOIN FETCH crd.certificationResultTestTools crtt "
+            + "LEFT OUTER JOIN FETCH crtt.testTool tt "
+            + "LEFT OUTER JOIN FETCH crd.certificationResultTestStandards crts "
+            + "LEFT OUTER JOIN FETCH crd.certificationResultAdditionalSoftware cras "
+            + "WHERE crd.deleted = false ";
+//            + "AND (crtd.deleted = false OR crtd.deleted IS NULL) "
```

Comment:
I'll take a look at this.  This code was commented out as part of the fix for the activity problem I was looking at this weekend.  When all the child data was deleted from the cert results, the cert result would no longer appear in the query results...

Reply:
This gets a little weird due to the left joins.  Since we are getting all of the cert results and associated collections in a single query, sometimes the all of the values from a table are null and we cannot evaluate the deleted flag.  I /feel/ like the commented code should work for the deleted and not found data, but it just doesn't seem to work.  This leads to having to check for null values when populating the domain collections in the CertificationResultService.  It's not the best solution, and now that I think about about it I'm not sure what we should do - we are using this same query "base" for all queries in that DAO, but I'm not sure we are properly handling deleted/null values.  I'm continuing to research.

Reply:
Understood that it is challenging. Maybe a better example is to look at the SurveillanceDAO. That behaves very similarly in that it is a big HQL statement that left joins all the possible pieces of surveillance and sometimes some of those things exist but not all of them. Those entities that get left-join-fetched in the surveillance HQL all have the `@Where(clause = "deleted <> 'true'")` annotation where they are referenced in the entity files. That code has been around for so long unchanged that I'm thinking it works and may be a good model to follow.